iocore/net/UnixNetVConnection.cc

```diff
@@ -990,7 +991,28 @@ UnixNetVConnection::load_buffer_and_write(int64_t towrite, MIOBufferAccessor &bu
 
     ink_assert(niov > 0);
     ink_assert(niov <= countof(tiovec));
-    r = socketManager.writev(con.fd, &tiovec[0], niov);
+
+#if DEBUG && (TCP_FASTOPEN == 0)
```

Comment:
`ink_assert` is only in debug mode, so that check is not useful. The other check should be blended in, e.g.
```
ink_assert(TCP_FASTOPEN != 0 || ! this->options.f_tcp_fastopen);
```